src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -446,12 +446,15 @@ public function update( $type, $item ) {
 			$allow_relaxed_file_ownership = true;
 		}
 
+		$is_debug = WP_DEBUG && WP_DEBUG_LOG;
 		if ( 'plugin' === $type ) {
 			$was_active = is_plugin_active( $upgrader_item );
-			error_log( '    Upgrading plugin ' . var_export( $item->slug, true ) . '...' );
+			if ( $is_debug ) {
+				error_log( '    Upgrading plugin ' . var_export( $item->slug, true ) . '...' );
+			}
```

Comment:
Note: An alternative syntax for this is: `$is_debug && error_log()` [3v4l](https://3v4l.org/9n2iD), which is supported by WPCS, but I went for the long version in case this is preferred.